### PR TITLE
chore: merge next into main for v5 release

### PR DESCRIPTION
**Pull request by jsumners**
For the same reasoning as given in https://github.com/pinojs/pino/pull/1947#issue-2260133148, we should just get v5 released.

***Important:*** once this PR is merged in:

1. We need to create a 4.x branch (actually, do this right before merging)
2. We need to issue new releases for all of the direct dependencies managed under the Fastify org and then update them in this repo
3. Should be able to issue the v5 release at this point

I'd say we get the thing out the door and then issue any sort of weblog post (i.e. we shouldn't block on the post being written first).

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by gurgunday** [comment]
I agree with releasing v5, _or at least a prerelease_, ASAP at this point

@mcollina told me he'd get to it in July

But by doing that direct merge to next from main, we just lost a few PRs:

https://github.com/fastify/fastify/pull/5419
https://github.com/fastify/fastify/pull/5444

**Comment by gurgunday** [comment]
Should I create a new PR for https://github.com/fastify/fastify/pull/5419?

**Comment by jsumners** [comment]
> Should I create a new PR for #5419?

No. Please just wait for me to mark this as ready for review.

**Comment by gurgunday** [comment]
Okay, just letting you know that if you restore https://github.com/fastify/fastify/pull/5419, some tests will _not_ pass because main merged https://github.com/fastify/fastify/pull/5439 and next needs https://github.com/fastify/fastify/pull/5444 to merge for those tests to pass, and like I said I can provide a separate PR that will fix all these at once to the current `next` branch, locally I have all the tests and coverage passing right now

**Comment by gurgunday** [comment]
It's fine, we should merge https://github.com/fastify/fastify/pull/5444 and it should be good

File: .github/workflows/benchmark.yml
Comment (gurgunday) [inline .github/workflows/benchmark.yml:19]: The benchmark workflows have a few more places where node 22 should be added, and <=16 should be removed

```suggestion
      MAIN-BENCH-20: ${{ steps.benchmark-main.outputs.BENCH_RESULT20 }}
      MAIN-BENCH-22: ${{ steps.benchmark-main.outputs.BENCH_RESULT22 }}
```

File: .github/workflows/benchmark-parser.yml
Comment (gurgunday) [inline .github/workflows/benchmark-parser.yml:19]: This one is the same as `benchmark.yml`

```suggestion
      MAIN-BENCH-20: ${{ steps.benchmark-main.outputs.BENCH_RESULT20 }}
      MAIN-BENCH-22: ${{ steps.benchmark-main.outputs.BENCH_RESULT22 }}
```

**Comment by mcollina** [comment]
I can have time to finalize v5 and issue releases for all the modules in July.

**Comment by mcollina** [comment]
This can land right now and we can  start shipping alphas.

**Comment by jsumners** [comment]
Yeah, I can't wait for July. My notifications are already unmanageable and the biggest reason is this ongoing v5 work.

File: package.json
Comment (gurgunday) [inline package.json:209]: This is no longer used and should be removed after https://github.com/fastify/fastify/pull/4450

Comment (jsumners) [reply package.json:209]: Cleaning up of dependencies can happen subsequent to this PR. This PR is about updating `main` for v5 and is already big enough.

File: .github/workflows/ci.yml
Comment (gurgunday) [inline .github/workflows/ci.yml:118]: Not that important but node 14 exclusions were removed so why not also remove 16?

```suggestion
```

**Review by gurgunday — APPROVED** [review]
lgtm

File: package.json
Comment (gurgunday) [reply package.json:209]: Sure, was just documenting for future reference

**Comment by mcollina** [comment]
As I said, I don't think I have the time deliver v5 before July. When I shipped v4 it took almost a week of full time work, and I can't simply fit into my calendar before then.

We can ship alphas/rcs before then.

We do need blog post and announcements etc, as it's also a marketing thing.

**Comment by mcollina** [comment]
I plan to issue another v4 release before landing this.

**Comment by jsumners** [comment]
> When I shipped v4 it took almost a week of full time work, and I can't simply fit into my calendar before then.

Right. None of us have time for that and really never will. We're just going to have to ship modules as we have time to get to them and/or as people indicate that they need the v5 compatible version.

I'm okay with getting the PR merged and going through alpha releases. At least getting `next => main` will hopefully allow for getting rid of the backlog of notifications.

**Comment by Eomm** [comment]
> We do need blog post and announcements etc, as it's also a marketing thing.

I can draft it 👍🏼

**Pull request merged**
